File: Lab1/Java/Lab1_POP/src/Stopper.java
class Stopper {
    public volatile boolean canStop = false;
    private int delayTime;

    public Stopper(int delayTime) {
        this.delayTime = delayTime;
    }

    public void setDelay() {
        try {
            Thread.sleep(delayTime * 1000);
            canStop = true;
        } catch (InterruptedException e) {
            e.printStackTrace();
        }
    }
}

File: Lab1/C#/Lab1POP/Lab1POP/Stopper.cs
using System;
using System.Threading;

namespace Lab1POP
{
    public class Stopper
    {
        public volatile bool canStop = false;
        private int delayTime;

        public Stopper(int delayTime)
        {
            this.delayTime = delayTime;
        }

        public void setDelay()
        {
            try
            {
                Thread.Sleep(delayTime * 1000);
                canStop = true;
            }
            catch(Exception e)
            {
                Console.WriteLine(e.Message);
            }
        }
    }
}
